feat(price create form): allow user to select a recent proof

dq18 opened the pull request:
### What

Allow user to select an existing proof when adding a price.

This is a WIP and only implemented for addSinglePrice. Please add your feedback to this PR on design and code and I will change / implement further.


### How
A new button allows to open a dialog with the last (max 10) proofs.
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/b2965b49-84c5-43c4-ab85-71897ebf41a8)
They will be displayed in a 5x2 grid.
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/e3650951-71fa-4b8d-86dc-05482c508fb2)
Selecting one will highlight it and show a confirmation dialog
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/a5a2cd4c-bd3e-4cc9-9bde-463a1b8f8f65)
If "No" is selected, user goes back to choosing the proof.
If "Yes", it closes both dialogs and displays the selected proof as is done for "Camera" or "Gallery" modes
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/dbe5ccf2-0be6-48fc-8054-542ee0dc30d1)

<hr>

PR simplified, no more confirmation dialog, works for both single & multiple price forms, rename "existing" to "recent"

teolemon commented:
@dq18 a number of things, more or less necessary, come to mind:
- is it possible to zoom on the proofs before selecting them ? 
- Adding a notion of time in the list of proofs might prove useful as well edge to edge line (Today, yesterday, etc ?)
- does it reselect the time of the proof in the price addition form ?
- should we allow linking proofs to places "asynchronously" (from your personal proof list, when you have free time), and thus prepopulate the place as well when selecting an existing proof ?

raphodn commented on src/components/ExistingProofDialog.vue:
```diff
@@ -0,0 +1,102 @@
+      <v-card-text>
+        <v-row>
+          <v-col cols="6" sm="6" md="3" v-for="proof in userProofList" :key="proof">
+            <ProofCard :proof="proof" :showProofFooter="false" :isSelectable="true" :isSelected="selectedProof === proof" @proofSelected="selectProof"></ProofCard>
```
i was thinking of doing a horizontal version of the ProofCard, but can be done after this PR & #375

raphodn replied:
but like toelemon mentions, maybe showing part or all of the footer could still be useful, to have context on each proof

raphodn commented:
Looks great ! Could almost be deployed right now :)
- don't hide the proof footer to have context ?
- "preuve téléversée" is not accurate (but non-blocking for merge)

dq18 commented:
Awesome feedback guys, I will get started to make the next version.
- I hid the footer for simplicity for a test version. I think that I can easily keep it but remove some chips (I think we only need the one with time).
- I'm not sure how to do zoom on each proof, but what I could do is to display it bigger when clicking on it (to be close to full screen) and have the dialog "Yes/No" somewhere at the bottom. WIll think a bit more about that.
- The time of the proof is kept. The dialog fetches the last 10 proofs with their info. I just reuse the date info for the price addition form. If you want a different behavior, let me know
- Having a separate proof upload page would be nice indeed, right now you have to go into any "add price" mode for that -> another PR :)
- "Proof uploaded" will be changed, along other hard-coded strings and other translations keys (will need some deep review though)

dq18 commented:
- Added feature for AddPriceMultiple
- Added ProofFooter in a minimal mode (just the time chip)
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/16026732-677d-42d5-a75d-1b70195b68cf)
- When clicking on a proof to choose, it now extends to the whole screen (in width because we keep ratio)
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/21193194/db4164bd-0bc5-4c22-9ec4-84cbd9547d5f)

I'm removing the WIP status, who validates this can merge too :)

teolemon reviewed (approved):
I'm validitating because I want this badly ;-) @raphodn you can merge if that's ok technically. 
We'll iterate on this one, until it becomes truly addictive.

raphodn commented on src/components/ExistingProofDialog.vue:
```diff
@@ -0,0 +1,99 @@
+</script>
+
+<style scoped>
+.d-flex {
```
why this style ? `d-flex` should work out of the box : https://vuetifyjs.com/en/styles/display/#display

dq18 replied:
indeed

raphodn commented on src/components/ExistingProofDialog.vue:
```diff
@@ -0,0 +1,99 @@
+        </v-row>
+    </v-card-text>
+    </v-card>
+    <v-bottom-sheet v-model="showConfirmationPopup">
```
currently the confirmation popups (delete a price, delete a proof) are `v-dialog`. I would have kept this instead of creating a new behavior with the `v-bottom-sheet` + complexifying the `ProofCard` when selected :thinking:

dq18 replied:
That was to be able to see the whole proof before validating. With a `v-dialog`, one can't see the proof before validating

raphodn replied:
v-dialog has a fullscreen option : https://vuetifyjs.com/en/api/v-dialog/#props-fullscreen

raphodn replied:
But honestly for a v1 I wouldn't even have a confirmation dialog (it's first and foremost for deletion, here it's... friction ?)
- worst case is that the user goes back to the "select a proof" 
- the user can zoom with 2 fingers
- 1 row of proofs on small screens

raphodn commented on src/components/ExistingProofDialog.vue:
```diff
@@ -0,0 +1,99 @@
+      <v-divider></v-divider>
+      <v-card-text>
+        <v-row>
+          <v-col cols="6" sm="6" md="3" v-for="proof in userProofList" :key="proof">
```
```suggestion
          <v-col cols="12" sm="6" md="3" v-for="proof in userProofList" :key="proof">
```
On small screens I think we should just show the list in 1 column. Also avoids having the zoomed image when selected, not really needed as the user can zoom already with 2 fingers

dq18 replied:
Alright

raphodn commented on src/components/ProofFooter.vue:
```diff
@@ -13,9 +13,9 @@
       
     </v-chip>
-    <PriceCountChip :count="proof.price_count" :withLabel="true"></PriceCountChip>
+    <PriceCountChip v-if="!reducedProofFooter" :count="proof.price_count" :withLabel="true"></PriceCountChip>
```
- why hide the price count ? seems a useful info to display
- and I'd rename `reducedProofFooter` to `hideProofDelete` (mecanism similar to other components)

dq18 replied:
the first `v-chip` for proof type is also hidden, so it would be named `hideProofTypeDelete`, but I suppose we should decide first what chip to keep before naming this boolean

raphodn replied:
why hide the proof type then ? :upside_down_face:

dq18 replied:
Taking a lot of space, and info not needed to in this case

raphodn replied:
I disagree, I wouldn't want to add GDPR proof here.
But in any case this is not important, you can keep it as is, it was just to simplify the logic

raphodn commented on src/views/AddPriceSingle.vue:
```diff
@@ -169,12 +173,15 @@
                 <p v-if="!proofFormFilled && !createProofLoading" class="text-red mt-2 mb-2">
                   <i>{{ $t('AddPriceSingle.PriceDetails.UploadProof') }}</i>
                 </p>
+                <p v-if="proofFormFilled && proofSelectedMessage" class="text-green mt-2 mb-2">
```
rather move this in the first `<p>` and add a v-if inside (avoids having 3 `<p>`)

dq18 replied:
Since there is already a v-if in the first `<p>` element, I don't think that would simplify. What I could do is group the 3 `<p>` together with a computed method

raphodn replied:
currently
![Screenshot from 2024-02-25 23-40-24](https://github.com/openfoodfacts/open-prices-frontend/assets/7147385/6224c8a1-8301-407d-ab98-58e0411604af)

what I meant
![Screenshot from 2024-02-25 23-41-32](https://github.com/openfoodfacts/open-prices-frontend/assets/7147385/1cab5432-fb00-453d-bf8f-04dbd4212dd7)

which one is simpler ? ^^

dq18 replied:
Alright, though I still think one <p> and a computed method is better :) But that can come in the future

raphodn commented on src/components/ProofCard.vue:
```diff
@@ -32,3 +50,20 @@ export default {
 </script>
+
+<style scoped>
+.highlighted {
```
never used ?

dq18 replied:
Not with the current implementation of showing the card in fullscreen when selected. However if I remove that implementation, do we want to highlight or just show the confirmation dialog?

raphodn replied:
neither ? proof simply selected on click

dq18 replied:
Alright. I commited the changes

raphodn commented:
merged ! made some last small changes : 
- make the dialog header "fixed"
- rename the dialog from "existing" to "recent" (to match with locations ; though it's very similar to "latest")
